Test check is too relaxed

### Describe the bug

When applying the following diff, the `test_component_provided_relation` does not pass:

```diff
diff --git a/gaphor/UML/tests/test_uml2_overrides.py b/gaphor/UML/tests/test_uml2_overrides.py
index b1964440c..f578e4466 100644
--- a/gaphor/UML/tests/test_uml2_overrides.py
+++ b/gaphor/UML/tests/test_uml2_overrides.py
@@ -32,7 +32,7 @@ def test_component_required_relation():
     component.clientDependency = usage
     usage.supplier = interface
 
-    assert interface in component.required
+    assert [interface] == component.required
 
 
 def test_component_provided_relation():
@@ -43,4 +43,4 @@ def test_component_provided_relation():
     component.interfaceRealization = realization
     realization.contract = interface
 
-    assert interface in component.provided
+    assert [interface] == component.provided
```

### To Reproduce
Apply the patch and run the test.

### Expected behavior
Test passes

### OS
Linux

### Version
Version of Gaphor: c22f3edef96e59407aed11f8e865c9d9686cf894

Sorry for not attaching the failure log in the original report, I now have access to the computer where I have this set up:

```
test_uml2_overrides.py ...F                                                                       [100%]

=============================================== FAILURES ================================================
___________________________________ test_component_provided_relation ____________________________________

    def test_component_provided_relation():
        realization = UML.InterfaceRealization()
        component = UML.Component()
        interface = UML.Interface()
    
        component.interfaceRealization = realization
        realization.contract = interface
    
>       assert [interface] == component.provided
E       assert [<gaphor.UML....fd136da0f5dc>] == [<gaphor.UML....fd136da0f5dc>]
E         Right contains one more item: <gaphor.UML.uml.Interface element 67b65e60-454d-11ee-ae9c-fd136da0f5dc>
E         Use -v to get more diff

test_uml2_overrides.py:46: AssertionError
```

To confirm that the issue is as I mentioned, I also asserted the length. So I now have the following diff:

```diff
diff --git a/gaphor/UML/tests/test_uml2_overrides.py b/gaphor/UML/tests/test_uml2_overrides.py
index b1964440c..bf6983c28 100644
--- a/gaphor/UML/tests/test_uml2_overrides.py
+++ b/gaphor/UML/tests/test_uml2_overrides.py
@@ -32,7 +32,7 @@ def test_component_required_relation():
     component.clientDependency = usage
     usage.supplier = interface
 
-    assert interface in component.required
+    assert [interface] == component.required
 
 
 def test_component_provided_relation():
@@ -43,4 +43,5 @@ def test_component_provided_relation():
     component.interfaceRealization = realization
     realization.contract = interface
 
-    assert interface in component.provided
+    assert len(component.provided) == 1
+    assert [interface] == component.provided
```

And this fails like this:

```
test_uml2_overrides.py ...F                                                                       [100%]

=============================================== FAILURES ================================================
___________________________________ test_component_provided_relation ____________________________________

    def test_component_provided_relation():
        realization = UML.InterfaceRealization()
        component = UML.Component()
        interface = UML.Interface()
    
        component.interfaceRealization = realization
        realization.contract = interface
    
>       assert len(component.provided) == 1
E       assert 2 == 1
E        +  where 2 = len([<gaphor.UML.uml.Interface element 7c6d382e-454d-11ee-ae9c-fd136da0f5dc>, <gaphor.UML.uml.Interface element 7c6d382e-454d-11ee-ae9c-fd136da0f5dc>])
E        +    where [<gaphor.UML.uml.Interface element 7c6d382e-454d-11ee-ae9c-fd136da0f5dc>, <gaphor.UML.uml.Interface element 7c6d382e-454d-11ee-ae9c-fd136da0f5dc>] = <gaphor.UML.uml.Component element 7c6d382d-454d-11ee-ae9c-fd136da0f5dc>.provided

test_uml2_overrides.py:46: AssertionError
```

--- a/gaphor/UML/tests/test_uml2_overrides.py
+++ b/gaphor/UML/tests/test_uml2_overrides.py
@@ -32,7 +32,7 @@ def test_component_required_relation():
     component.clientDependency = usage
     usage.supplier = interface
 
-    assert interface in component.required
+    assert [interface] == component.required
 
 
 def test_component_provided_relation():
@@ -43,4 +43,4 @@ def test_component_provided_relation():
     component.interfaceRealization = realization
     realization.contract = interface
 
-    assert interface in component.provided
+    assert [interface] == component.provided

--- a/gaphor/UML/umloverrides.py
+++ b/gaphor/UML/umloverrides.py
@@ -113,10 +113,10 @@ def component_provided(self: uml.Component) -> list[uml.Realization]:
     realizations = _pr_interface_deps(self, uml.Realization)
 
     # realizing classifiers realizations
-    # this generator of generators, so flatten it later
+    # this is a generator of generators, so flatten it later
     rc_realizations = _pr_rc_interface_deps(self, uml.Realization)
 
-    return list(itertools.chain(implementations, realizations, *rc_realizations))
+    return list(set(itertools.chain(implementations, realizations, *rc_realizations)))
 
 
 uml.Component.provided = property(component_provided, doc=component_provided.__doc__)

